Source/HTTP/Curl/CurlProvider.cpp

```diff
@@ -19,8 +19,10 @@ HRESULT Internal_InitializeHttpPlatform(HCInitArgs* args, PerformEnv& performEnv
 
 void Internal_CleanupHttpPlatform(HC_PERFORM_ENV* performEnv) noexcept
 {
-    // HC_PERFORM_ENV created with custom deleter - ~HC_PERFORM_ENV needs to be explicitly invoked.
-    performEnv->~HC_PERFORM_ENV();
+    // HC_PERFORM_ENV created with custom deleter - HC_PERFORM_ENV needs to be destroyed and cleaned up explicitly.
+    http_stl_allocator<HC_PERFORM_ENV> a{};
+    std::allocator_traits<http_stl_allocator<HC_PERFORM_ENV>>::destroy(a, std::addressof(*performEnv));
```

Comment:
why the deref / std::addressof dance?

Reply:
looks unnecessary (pulled this from winHttp implementation of Cleanup)